server/src/test/java/org/opensearch/action/search/GetAllPitNodesResponseTests.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.Version;
+import org.opensearch.action.FailedNodeException;
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.core.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.transport.TransportException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.emptySet;
+
+public class GetAllPitNodesResponseTests extends OpenSearchTestCase {
```

Comment:
Thanks @bugmakerrrrrr but I think we need integration test that verifies this claim

```
Today, we haven't reconstructed pit infos when deserialize GetAllPitNodesResponse, which causes corresponding field is empty.
```

Reply:
I usually write related integration tests through TransportClient, but this has been deprecated in OS. Maybe you can give me a related example?

Reply:
Would this one be helpful: `CreatePitSingleNodeTests` ?

Reply:
Thanks, but I don't think the response of either singleNodeTests or multiNodeTests will go through serialization. Do I miss something😂?

Reply:
For `multiNodeTests` it definitely should, since we would be serializing the response from multiple transport nodes on coordinator side

Reply:
I think this is the reason why the previous test did not report an error.

> Yes, but we just deserialize nodeResponse on coordinator, not nodesResponse. And the coordinator basically constructs a response through `XContentBuilder` and then returns it to the client.

I think this is the reason why the previous test did not report an error.

Reply:
So the `GetAllPitNodesResponse` seems to be only taking part in whit `PitSegmentsAction` is called (`cat` API if I am not mistaken):

```
PitSegmentsRequest request = new PitSegmentsRequest("_all").indices("*");
ActionFuture<IndicesSegmentResponse> future = client().execute(PitSegmentsAction.INSTANCE, request2);
future.get();
```

However, I am not able to figure out under what circumstances the constructor in question is being called:

```
                @Override
                public GetAllPitNodesResponse read(StreamInput in) throws IOException {
                    return new GetAllPitNodesResponse(in);
                }
```

I am wondering how the issue was discovered in first place?

Reply:
> However, I am not able to figure out under what circumstances the constructor in question is being called:
> 
> ```
>                 @Override
>                 public GetAllPitNodesResponse read(StreamInput in) throws IOException {
>                     return new GetAllPitNodesResponse(in);
>                 }
> ```

I think this method will not be called.

> I am wondering how the issue was discovered in first place?

I found it when reading the code.

Reply:
> > However, I am not able to figure out under what circumstances the constructor in question is being called:
> > ```
> >                 @Override
> >                 public GetAllPitNodesResponse read(StreamInput in) throws IOException {
> >                     return new GetAllPitNodesResponse(in);
> >                 }
> > ```
> 
> I think this method will not be called.
> 
> > I am wondering how the issue was discovered in first place?
> 
> I found it when reading the code.

Could you please fix DCO and we could merge it in the current shape, it is an issue in any case

Reply:
done